galley/cmd/galley/cmd/root.go

```diff
@@ -48,21 +64,94 @@ func GetRootCmd(args []string, printf, fatalf shared.FormatFn) *cobra.Command {
 			if len(args) > 0 {
 				return fmt.Errorf("%q is an invalid argument", args[0])
 			}
-
 			return log.Configure(loggingOptions)
 		},
+		Run: func(cmd *cobra.Command, args []string) {
+
+			serverArgs.KubeConfig = flags.kubeConfig
+			serverArgs.ResyncPeriod = flags.resyncPeriod
+			serverArgs.CACertificateFile = validationArgs.CACertFile
+			serverArgs.KeyFile = validationArgs.KeyFile
+			serverArgs.CertificateFile = validationArgs.CertFile
+
+			if livenessProbeOptions.IsValid() {
+				livenessProbeController = probe.NewFileController(&livenessProbeOptions)
+			}
+			if readinessProbeOptions.IsValid() {
+				readinessProbeController = probe.NewFileController(&readinessProbeOptions)
+
+			}
+
+			if !serverArgs.EnableServer && !validationArgs.EnableValidation {
+				fatalf("Galley must be running under at least one mode: server or validation")
+			}
+
+			if serverArgs.EnableServer {
+				go server.RunServer(serverArgs, printf, fatalf, livenessProbeController, readinessProbeController)
+			}
+			if validationArgs.EnableValidation {
+				go validation.RunValidation(validationArgs, printf, fatalf, flags.kubeConfig, livenessProbeController, readinessProbeController)
+			}
+
+			galleyStop := make(chan struct{})
+			go server.StartProbeCheck(livenessProbeController, readinessProbeController, galleyStop)
```

Comment:
`StartProbeCheck` doesn't need to be run as a goroutine as it starts goroutines of its own and is non-blocking.

Reply:
done